### DISCUSSION TITLE
Remix + Lazy Route Discovery

### DISCUSSION THREAD

**Discussion by wilcoxmd** [remix-run/remix#10004] [Q&A]
Hi! 

I'm trying to figure out if we can use [Lazy Route Discovery](https://remix.run/blog/fog-of-war) with custom/remote lazy route loading in Remix, or if that's a react router only option. I'm having a hard time finding how I can try out something like `unstable_patchRoutesOnMiss` from within Remix's configuration.  

Thanks!

**Comment by jrestall** [comment]
It doesn't seem to be exposed in Remix since it's used internally to implement Remix's lazy route discovery. Exposing it would risk impacting that functionality since it's built on `unstable_patchRoutesOnNavigation`.

https://github.com/remix-run/remix/blob/58d167d155aef297c29e3d9587068309f4977a14/packages/remix-react/browser.tsx#L313

You would have to use `patch-package` or `pnpm patch` to add a hook if you want to add your own custom logic.
